diff -uNr rpm-4.5/autodeps/darwin.prov rpm-4.5-new/autodeps/darwin.prov
--- rpm-4.5/autodeps/darwin.prov	2005-01-04 09:41:39.000000000 -0800
+++ rpm-4.5-new/autodeps/darwin.prov	2007-06-13 10:52:54.000000000 -0700
@@ -5,7 +5,7 @@
 # This script reads filenames from STDIN and outputs any relevant provides
 # information that needs to be included in the package.
 
-filelist=$(sed -n -e '/\.dylib/p' -e '/\.so/p' -e '/\.bundle/p' | sort | uniq | xargs file -L 2>/dev/null | grep "Mach-O.*ppc" | cut -d: -f1)
+filelist=$(sed -n -e '/\.dylib/p' -e '/\.so/p' -e '/\.bundle/p' -e '/\.jnilib/p' | sort | uniq | xargs file -L 2>/dev/null | grep "Mach-O.*" | cut -d: -f1)
 
 for f in $filelist; do
     libname=$(basename $f | sed -e 's;\..*;;')
diff -uNr rpm-4.5/autodeps/darwin.req rpm-4.5-new/autodeps/darwin.req
--- rpm-4.5/autodeps/darwin.req	2005-01-04 09:41:39.000000000 -0800
+++ rpm-4.5-new/autodeps/darwin.req	2007-06-13 10:55:52.000000000 -0700
@@ -15,7 +15,7 @@
 	| sed -n -e '/ (compatibility version .* current version .*)/p' \
 	| sed -e 's/ (compatibility version .* current version .*)//'
     fi
-done | sort -u | sed "s/['\"]/\\\&/g" | xargs -n 1 basename | sort -u
+done | sort -u | sed "s/['\"]/\\\&/g" | grep -v -E '^(/System/Library/|/Library/|/lib/|/usr/lib/|/usr/local/lib/)' | xargs -n 1 basename | sort -u
 
 for f in $scriptlist; do
     if [ -x $f ]; then
diff -uNr rpm-4.5/lib/rpmrc.c rpm-4.5-new/lib/rpmrc.c
--- rpm-4.5/lib/rpmrc.c	2007-05-25 10:36:02.000000000 -0700
+++ rpm-4.5-new/lib/rpmrc.c	2007-06-13 11:03:40.000000000 -0700
@@ -585,7 +585,7 @@
     addMacro(NULL, "___build_pre", NULL, ___build_pre, RMIL_DEFAULT);
 
     setVarDefault(-1,			"_topdir",
-		"/usr/src/rpm",		"%{_usr}/src/rpm");
+		"/usr/src/rpm",		"%{_usr}/var/rpm-src");
     setVarDefault(-1,			"_tmppath",
 		"/var/tmp",		"%{_var}/tmp");
     setVarDefault(-1,			"_dbpath",
diff -uNr rpm-4.5/rpmdb/rpmdb.c rpm-4.5-new/rpmdb/rpmdb.c
--- rpm-4.5/rpmdb/rpmdb.c	2007-05-25 10:36:33.000000000 -0700
+++ rpm-4.5-new/rpmdb/rpmdb.c	2007-06-13 10:58:01.000000000 -0700
@@ -3868,7 +3868,7 @@
 
     /* Remove /etc/rpm/macros.db1 configuration file if db3 rebuilt. */
     if (rc == 0 && _newdbapi == 1 && _olddbapi == 3) {
-	const char * mdb1 = "/etc/rpm/macros.db1";
+	const char * mdb1 = "@FINKPREFIX@/etc/rpm/macros.db1";
 	struct stat st;
 	if (!stat(mdb1, &st) && S_ISREG(st.st_mode) && !unlink(mdb1))
 	    rpmMessage(RPMMESS_DEBUG,
diff -uNr rpm-4.5/rpmio/fts.c rpm-4.5-new/rpmio/fts.c
--- rpm-4.5/rpmio/fts.c	2007-05-25 10:36:36.000000000 -0700
+++ rpm-4.5-new/rpmio/fts.c	2007-06-13 10:52:54.000000000 -0700
@@ -34,6 +34,7 @@
 static char sccsid[] = "@(#)fts.c	8.6 (Berkeley) 8/14/94";
 #endif /* LIBC_SCCS and not lint */
 
+#include "system.h"
 #if defined(_LIBC)
 #include <sys/param.h>
 #include <include/sys/stat.h>
@@ -70,7 +71,6 @@
 #   define stat64              stat
 #   define __fxstat64(_stat_ver, _fd, _sbp)    fstat((_fd), (_sbp))
 #endif
-#include "system.h"
 #include "fts.h"
 #include "rpmio.h"
 #include "rpmurl.h"
diff -uNr rpm-4.5/rpmio/macro.c rpm-4.5-new/rpmio/macro.c
--- rpm-4.5/rpmio/macro.c	2007-05-25 10:36:36.000000000 -0700
+++ rpm-4.5-new/rpmio/macro.c	2007-06-13 10:57:49.000000000 -0700
@@ -2419,7 +2419,7 @@
 
 #if defined(EVAL_MACROS)
 
-char *rpmMacrofiles = "/usr/lib/rpm/macros:/etc/rpm/macros:~/.rpmmacros";
+char *rpmMacrofiles = "@FINKPREFIX@/etc/rpm/macros:~/.rpmmacros";
 
 int
 main(int argc, char *argv[])
diff -uNr rpm-4.5/rpmio/rpmio.h rpm-4.5-new/rpmio/rpmio.h
--- rpm-4.5/rpmio/rpmio.h	2007-05-25 12:00:39.000000000 -0700
+++ rpm-4.5-new/rpmio/rpmio.h	2007-06-13 10:52:54.000000000 -0700
@@ -664,7 +664,11 @@
  * @param prompt	prompt string
  * @return		password
  */
+#ifdef NO_EXTERN
 char * (*Getpass) (const char * prompt)
+#else
+extern char * (*Getpass) (const char * prompt)
+#endif
 	/*@*/;
 char * _GetPass (const char * prompt)
 	/*@*/;
