diff -uNr Mesa-6.5.3/Makefile Mesa-6.5.3-patched/Makefile
--- Mesa-6.5.3/Makefile	2007-04-27 09:54:33.000000000 -0400
+++ Mesa-6.5.3-patched/Makefile	2007-07-17 23:38:02.000000000 -0400
@@ -23,7 +23,6 @@
 realclean:
 	touch $(TOP)/configs/current
 	$(MAKE) clean
-	-rm -rf lib*
 	-rm -f $(TOP)/configs/current
 	-rm -f `find . -name \*.o`
 	-rm -f `find . -name \*.a`
diff -uNr Mesa-6.5.3/bin/mklib Mesa-6.5.3-patched/bin/mklib
--- Mesa-6.5.3/bin/mklib	2007-03-27 12:28:55.000000000 -0400
+++ Mesa-6.5.3-patched/bin/mklib	2007-07-17 23:26:01.000000000 -0400
@@ -570,10 +570,11 @@
                 OPTS="${ARCHOPT} -bundle -multiply_defined suppress"
             else
 		LIBSUFFIX="dylib"
-                OPTS="${ARCHOPT} -dynamiclib -multiply_defined suppress -current_version ${MAJOR}.${MINOR}.0 -compatibility_version ${MAJOR}.${MINOR}.0 -install_name lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
+                OPTS="${ARCHOPT} -dynamiclib -multiply_defined suppress -headerpad_max_install_names -current_version ${MAJOR}.${MINOR}.0 -compatibility_version ${MAJOR}.${MINOR}.0 -install_name @PREFIX@/lib/lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
             fi
-            LINKNAME="lib${LIBNAME}.${LIBSUFFIX}"
-            LIBNAME="lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
+            LINKNAME="lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
+            LINKNAME2="lib${LIBNAME}.${LIBSUFFIX}"
+            LIBNAME="lib${LIBNAME}.${MAJOR}.${MINOR}.${LIBSUFFIX}"
 
 	    # examine first object to determine ABI
     	    set ${OBJECTS}
@@ -599,7 +600,8 @@
             echo "mklib: Making Darwin shared library: " ${LIBNAME}
             ${LINK} ${OPTS} -o ${LIBNAME} ${OBJECTS} ${DEPS}
             ln -s ${LIBNAME} ${LINKNAME}
-            FINAL_LIBS="${LIBNAME} ${LINKNAME}"
+            ln -s ${LINKNAME} ${LINKNAME2}
+            FINAL_LIBS="${LIBNAME} ${LINKNAME} ${LINKNAME2}"
         fi
         ;;
 
diff -uNr Mesa-6.5.3/configs/darwin Mesa-6.5.3-patched/configs/darwin
--- Mesa-6.5.3/configs/darwin	2006-12-07 18:16:31.000000000 -0500
+++ Mesa-6.5.3-patched/configs/darwin	2007-07-17 23:10:12.000000000 -0400
@@ -7,8 +7,8 @@
 # Compiler and flags
 CC = cc
 CXX = cc
-CFLAGS = -I/usr/X11R6/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
-CXXFLAGS = -I/usr/X11R6/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
+CFLAGS = -I@PREFIX@/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -dynamic -Ddarwin
+CXXFLAGS = -I@PREFIX@/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -dynamic -Ddarwin
 
 # Library names (actual file names)
 GL_LIB_NAME = libGL.dylib
@@ -17,12 +17,12 @@
 GLW_LIB_NAME = libGLw.dylib
 OSMESA_LIB_NAME = libOSMesa.dylib
 
-GL_LIB_DEPS = -L/usr/X11R6/lib -lX11 -lXext -lm -lpthread
+GL_LIB_DEPS = -L@PREFIX@/lib -lX11 -lXext -lm -lpthread
 OSMESA_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL
 GLU_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL
-GLUT_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL -lGLU -L/usr/X11R6/lib -lX11 -lXmu -lXi -lXext
-GLW_LIB_DEPS = -L/usr/X11R6/lib -lX11 -lXt $(TOP)/lib/GL.dylib
-APP_LIB_DEPS = -L$(TOP)/lib -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L/usr/X11R6/lib -lX11 -lXmu -lXt -lXi -lm
+GLUT_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL -lGLU -L@PREFIX@/lib -lX11 -lXmu -lXi -lXext
+GLW_LIB_DEPS = -L@PREFIX@/lib -lX11 -lXt $(TOP)/lib/GL.dylib
+APP_LIB_DEPS = -L$(TOP)/lib -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L@PREFIX@/lib -lX11 -lXmu -lXt -lXi -lm
 
 # omit glw lib for now:
 SRC_DIRS = mesa glu glut/glx
diff -uNr Mesa-6.5.3/configs/darwin-static Mesa-6.5.3-patched/configs/darwin-static
--- Mesa-6.5.3/configs/darwin-static	2006-12-07 18:16:31.000000000 -0500
+++ Mesa-6.5.3-patched/configs/darwin-static	2007-07-17 23:10:12.000000000 -0400
@@ -5,8 +5,8 @@
 CONFIG_NAME = darwin-static
 
 # Compiler and flags
-CFLAGS = -I/usr/X11R6/include -O3 -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
-CXXFLAGS = -I/usr/X11R6/include -O3 -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
+CFLAGS = -I@PREFIX@/include -O3 -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -dynamic -Ddarwin
+CXXFLAGS = -I@PREFIX@/include -O3 -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -dynamic -Ddarwin
 MKLIB_OPTIONS = -static
 
 # Library names (actual file names)
@@ -21,4 +21,4 @@
 GLU_LIB_DEPS = 
 GLUT_LIB_DEPS = 
 GLW_LIB_DEPS = 
-APP_LIB_DEPS = -Wl,-search_paths_first -L$(TOP)/lib -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L/usr/X11R6/lib -lX11 -lXmu -lXt -lXi -lstdc++ -lgcc_s.1 -lm
+APP_LIB_DEPS = -Wl,-search_paths_first -L$(TOP)/lib -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L@PREFIX@/lib -lX11 -lXmu -lXt -lXi -lstdc++ -lgcc_s.1 -lm
diff -uNr Mesa-6.5.3/configs/darwin-static-x86ppc Mesa-6.5.3-patched/configs/darwin-static-x86ppc
--- Mesa-6.5.3/configs/darwin-static-x86ppc	2006-12-07 18:16:31.000000000 -0500
+++ Mesa-6.5.3-patched/configs/darwin-static-x86ppc	2007-07-17 23:10:12.000000000 -0400
@@ -6,9 +6,9 @@
 
 # Compiler and flags
 CFLAGS = -arch ppc -arch i386 -isysroot /Developer/SDKs/MacOSX10.4u.sdk \
-         -I/usr/X11R6/include -O3 -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
+         -I@PREFIX@/include -O3 -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -dynamic -Ddarwin
 CXXFLAGS = -arch ppc -arch i386 -isysroot /Developer/SDKs/MacOSX10.4u.sdk \
-           -I/usr/X11R6/include -O3 -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
+           -I@PREFIX@/include -O3 -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -dynamic -Ddarwin
 MKLIB_OPTIONS = -static -archopt "-isysroot /Developer/SDKs/MacOSX10.4u.sdk"
 
 # Library names (actual file names)
@@ -23,4 +23,4 @@
 GLU_LIB_DEPS = 
 GLUT_LIB_DEPS = 
 GLW_LIB_DEPS = 
-APP_LIB_DEPS = -Wl,-search_paths_first -L$(TOP)/lib -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L/usr/X11R6/lib -lX11 -lXmu -lXt -lXi -lstdc++ -lgcc_s.1 -lm
+APP_LIB_DEPS = -Wl,-search_paths_first -L$(TOP)/lib -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L@PREFIX@/lib -lX11 -lXmu -lXt -lXi -lstdc++ -lgcc_s.1 -lm
diff -uNr Mesa-6.5.3/configs/darwin-x86ppc Mesa-6.5.3-patched/configs/darwin-x86ppc
--- Mesa-6.5.3/configs/darwin-x86ppc	2006-12-07 18:16:31.000000000 -0500
+++ Mesa-6.5.3-patched/configs/darwin-x86ppc	2007-07-17 23:10:12.000000000 -0400
@@ -8,9 +8,9 @@
 CC = cc
 CXX = cc
 CFLAGS = -arch ppc -arch i386 -isysroot /Developer/SDKs/MacOSX10.4u.sdk \
-         -I/usr/X11R6/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
+         -I@PREFIX@/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -dynamic -Ddarwin
 CXXFLAGS = -arch ppc -arch i386 -isysroot /Developer/SDKs/MacOSX10.4u.sdk \
-         -I/usr/X11R6/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
+         -I@PREFIX@/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -dynamic -Ddarwin
 
 MKLIB_OPTIONS = -archopt "-isysroot /Developer/SDKs/MacOSX10.4u.sdk"
 
@@ -21,12 +21,12 @@
 GLW_LIB_NAME = libGLw.dylib
 OSMESA_LIB_NAME = libOSMesa.dylib
 
-GL_LIB_DEPS = -L/usr/X11R6/lib -lX11 -lXext -lm -lpthread
+GL_LIB_DEPS = -L@PREFIX@/lib -lX11 -lXext -lm -lpthread
 OSMESA_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL
 GLU_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL
-GLUT_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL -lGLU -L/usr/X11R6/lib -lX11 -lXmu -lXi -lXext
-GLW_LIB_DEPS = -L/usr/X11R6/lib -lX11 -lXt $(TOP)/lib/GL.dylib
-APP_LIB_DEPS = -L$(TOP)/lib -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L/usr/X11R6/lib -lX11 -lXmu -lXt -lXi -lm
+GLUT_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL -lGLU -L@PREFIX@/lib -lX11 -lXmu -lXi -lXext
+GLW_LIB_DEPS = -L@PREFIX@/lib -lX11 -lXt $(TOP)/lib/GL.dylib
+APP_LIB_DEPS = -L$(TOP)/lib -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L@PREFIX@/lib -lX11 -lXmu -lXt -lXi -lm
 
 # omit glw lib for now:
 SRC_DIRS = mesa glu glut/glx
