diff --git a/src/3rdparty/webkit/WebKit/qt/declarative/declarative.pro b/src/3rdparty/webkit/WebKit/qt/declarative/declarative.pro
index 526cf06..0d3bd9e 100644
--- a/src/3rdparty/webkit/WebKit/qt/declarative/declarative.pro
+++ b/src/3rdparty/webkit/WebKit/qt/declarative/declarative.pro
@@ -8,6 +8,8 @@ win32|mac:!wince*:!win32-msvc:!macx-xcode:CONFIG += debug_and_release
 
 isEmpty(OUTPUT_DIR): OUTPUT_DIR = ../../..
 
+INCLUDEPATH += $$OUTPUT_DIR/WebKit/qt/Api
+
 QMLDIRFILE = $${_PRO_FILE_PWD_}/qmldir
 copy2build.input = QMLDIRFILE
 CONFIG(QTDIR_build) {
