diff -Nurd kdeedu-4.3.90/cantor/src/CMakeLists.txt kdeedu-4.3.90-new/cantor/src/CMakeLists.txt
--- kdeedu-4.3.90/cantor/src/CMakeLists.txt	2009-12-10 18:14:14.000000000 -0500
+++ kdeedu-4.3.90-new/cantor/src/CMakeLists.txt	2010-01-14 23:51:22.000000000 -0500
@@ -3,6 +3,9 @@
 #########################################################################
 add_subdirectory(lib)
 
+set(GENERIC_LIB_VERSION "0.0.1")
+set(GENERIC_LIB_SOVERSION "1")
+
 include_directories( lib ${CMAKE_CURRENT_BINARY_DIR}/lib)
 if(LIBSPECTRE_FOUND)
    include_directories(${LIBSPECTRE_INCLUDE_DIR})
@@ -16,6 +19,8 @@
 target_link_libraries( cantor_config ${KDE4_KDEUI_LIBS} ${KDE4_KPARTS_LIBS} ${KDE4_KNEWSTUFF2_LIBS} )
 install( TARGETS cantor_config  ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
+set_target_properties( cantor_config PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
+
 set(cantor_SRCS
    main.cpp
    cantor.cpp
