diff --git a/mkspecs/common/mac-g++.conf b/mkspecs/common/mac-g++.conf
index a10035a..e31c726 100644
--- a/mkspecs/common/mac-g++.conf
+++ b/mkspecs/common/mac-g++.conf
@@ -5,7 +5,7 @@
 include(mac.conf)
 
 QMAKE_COMPILER_DEFINES  += __APPLE__ __GNUC__
-QMAKE_CFLAGS		+= -pipe
+QMAKE_CFLAGS		+= -pipe -I@FINKPREFIX@/lib/system-openssl/include
 QMAKE_CFLAGS_DEPS	+= -M
 QMAKE_CFLAGS_WARN_ON	+= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	+= -w
@@ -52,7 +52,7 @@ QMAKE_LINK		 = $$QMAKE_CXX
 QMAKE_LINK_SHLIB	 = $$QMAKE_CXX
 QMAKE_LINK_C		 = $$QMAKE_CC
 QMAKE_LINK_C_SHLIB	 = $$QMAKE_CC
-QMAKE_LFLAGS		+= -headerpad_max_install_names
+QMAKE_LFLAGS		+= -headerpad_max_install_names -L@FINKPREFIX@/lib/system-openssl/lib
 QMAKE_LFLAGS_RELEASE	+=
 QMAKE_LFLAGS_RELEASE_WITH_DEBUGINFO += -g -gdwarf-2
 QMAKE_LFLAGS_DEBUG	+=
diff --git a/mkspecs/darwin-g++/qmake.conf b/mkspecs/darwin-g++/qmake.conf
index ad1f163..ab1de5c 100644
--- a/mkspecs/darwin-g++/qmake.conf
+++ b/mkspecs/darwin-g++/qmake.conf
@@ -15,7 +15,7 @@ QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		=
 QMAKE_YACC		= yacc
 QMAKE_YACCFLAGS		= -d
-QMAKE_CFLAGS		= -pipe
+QMAKE_CFLAGS		= -pipe -I@FINKPREFIX@/lib/system-openssl/include
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	= -w
@@ -55,8 +55,7 @@ QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
 
 QMAKE_LINK		= c++
 QMAKE_LINK_SHLIB	= c++
-QMAKE_LFLAGS		+= -headerpad_max_install_names
-QMAKE_LFLAGS		=
+QMAKE_LFLAGS		+= -headerpad_max_install_names -L@FINKPREFIX@/lib/system-openssl/lib
 QMAKE_LFLAGS_RELEASE	=
 QMAKE_LFLAGS_DEBUG	=
 QMAKE_LFLAGS_APP        = -prebind
@@ -69,7 +68,6 @@ QMAKE_LFLAGS_VERSION    = -current_version$${LITERAL_WHITESPACE}
 QMAKE_LFLAGS_COMPAT_VERSION = -compatibility_version$${LITERAL_WHITESPACE}
 
 QMAKE_LFLAGS_RPATH	=
-QMAKE_RPATH		=
 QMAKE_FIX_RPATH		= install_name_tool -id 
 
 QMAKE_LIBS_DYNLOAD	=
diff --git a/configure-for-fink.sh b/configure-for-fink.sh
index 3005bd6..002f715 100755
--- a/configure-for-fink.sh
+++ b/configure-for-fink.sh
@@ -16,26 +16,10 @@ PATH="$QTDIR/bin:$FINKPREFIX/lib/freetype219/bin:$PATH"
 
 EXTRA_ARGS=""
 
-[ -z "$CC" ]  && export CC=gcc
-[ -z "$CXX" ] && export CXX=g++
-export PLATFORM_POSTFIX=""
-
-if [ -x /usr/bin/gcc-4.2 ]; then
-	export CC=gcc-4.2
-	export CXX=g++-4.2
-	export PLATFORM_POSTFIX="42"
-fi 
-
-if [ -x "$FINKPREFIX/bin/ccache" ]; then
-	export CC="$FINKPREFIX/bin/ccache $CC"
-	export CXX="$FINKPREFIX/bin/ccache $CXX"
-	export CCACHE_DIR="/tmp/ccache-qt4"
-fi
-
 if [ "$PKGNAME" = "qt4-x11" ]; then
-	EXTRA_ARGS="-x11 -platform darwin-g++$PLATFORM_POSTFIX -xplatform darwin-g++$PLATFORM_POSTFIX"
+	EXTRA_ARGS="-x11 -platform darwin-g++ -xplatform darwin-g++"
 else
-	EXTRA_ARGS="-platform macx-g++$PLATFORM_POSTFIX -xplatform macx-g++$PLATFORM_POSTFIX"
+	EXTRA_ARGS="-platform macx-g++ -xplatform macx-g++"
 fi
 
 case `sw_vers -productVersion` in
