diff -Nurd rubygems-1.3.5/lib/rubygems/commands/setup_command.rb rubygems-1.3.5-new/lib/rubygems/commands/setup_command.rb
--- rubygems-1.3.5/lib/rubygems/commands/setup_command.rb	2009-06-25 20:43:45.000000000 -0400
+++ rubygems-1.3.5-new/lib/rubygems/commands/setup_command.rb	2010-02-08 10:08:19.000000000 -0500
@@ -117,7 +117,7 @@
 
     say "RubyGems #{Gem::VERSION} installed"
 
-    install_rdoc
+    install_rdoc(install_destdir)
 
     say
     if @verbose then
@@ -228,8 +228,8 @@
     end
   end
 
-  def install_rdoc
-    gem_doc_dir = File.join Gem.dir, 'doc'
+  def install_rdoc(install_destdir)
+    gem_doc_dir = File.join install_destdir, Gem.dir, '/doc'
     rubygems_name = "rubygems-#{Gem::RubyGemsVersion}"
     rubygems_doc_dir = File.join gem_doc_dir, rubygems_name
 
@@ -237,7 +237,7 @@
        (not File.exist? rubygems_doc_dir or
         File.writable? rubygems_doc_dir) then
       say "Removing old RubyGems RDoc and ri" if @verbose
-      Dir[File.join(Gem.dir, 'doc', 'rubygems-[0-9]*')].each do |dir|
+      Dir[File.join(gem_doc_dir, 'rubygems-[0-9]*')].each do |dir|
         rm_rf dir
       end
 
diff -Nurd rubygems-1.3.5/lib/rubygems/defaults.rb rubygems-1.3.5-new/lib/rubygems/defaults.rb
--- rubygems-1.3.5/lib/rubygems/defaults.rb	2009-06-22 18:54:36.000000000 -0400
+++ rubygems-1.3.5-new/lib/rubygems/defaults.rb	2010-02-08 10:01:05.000000000 -0500
@@ -20,13 +20,8 @@
     if defined? RUBY_FRAMEWORK_VERSION then
       File.join File.dirname(ConfigMap[:sitedir]), 'Gems',
                 ConfigMap[:ruby_version]
-    # 1.9.2dev reverted to 1.8 style path
-    elsif RUBY_VERSION > '1.9' and RUBY_VERSION < '1.9.2' then
-      File.join(ConfigMap[:libdir], ConfigMap[:ruby_install_name], 'gems',
-                ConfigMap[:ruby_version])
     else
-      File.join(ConfigMap[:libdir], ruby_engine, 'gems',
-                ConfigMap[:ruby_version])
+      File.join('@PREFIX@', 'var', 'lib', 'gems', ConfigMap[:ruby_version])
     end
   end
 
diff -Nurd rubygems-1.3.5/lib/rubygems/source_index.rb rubygems-1.3.5-new/lib/rubygems/source_index.rb
--- rubygems-1.3.5/lib/rubygems/source_index.rb	2009-06-22 19:00:52.000000000 -0400
+++ rubygems-1.3.5-new/lib/rubygems/source_index.rb	2010-02-08 10:01:40.000000000 -0500
@@ -313,7 +313,7 @@
                     end
     else
       version_requirement = platform_only || Gem::Requirement.default
-      gem_pattern = /#{gem_pattern}/i
+      gem_pattern = /^#{gem_pattern}/i
     end
 
     unless Gem::Requirement === version_requirement then
